### PR TITLE
json2junit.py: Add script to convert Yocto's JSON into jUnit

### PR DESCRIPTION
**Pull request by mardy**
Since Yocto `thud`, test reports are delivered in the form of a single
JSON file. This script takes it as input and generates jUnit XML files,
similarly to how things worked before `thud`.

### CODE REVIEW

File: yocto/json2junit.py [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+#
+# THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL
+# WARRANTIES WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED
+# WARRANTIES OF MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE AUTHOR
+# BE LIABLE FOR ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES
+# OR ANY DAMAGES WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS,
+# WHETHER IN AN ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION,
+# ARISING OUT OF OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS
+# SOFTWARE.
+#
```
Comment (fhajredini) [inline yocto/json2junit.py]: Include usage

**Comment by fhajredini** [comment]
Besides the usage, looks good.

File: yocto/json2junit.py (hunk 1 above)
Comment (mardy) [reply yocto/json2junit.py]: Sorry Fisnik, could you clarify what you mean?

Comment (fhajredini) [reply yocto/json2junit.py]: If you check another script as an example, such as [smoke-test script](https://github.com/Pelagicore/vagrant-cookbook/blob/master/yocto/runqemu-smoke-test.sh), we have there a line that describes how to call the script. 

# Usage: smoke-test-images.sh <yoctodir> <images>

I am looking for something similar.

Comment (mardy) [reply yocto/json2junit.py]: Ah, silly me! Sure, I'll add it :-)

**Comment by mardy** [comment]
> How do you use this guy?

In `thud` and later, Yocto smoke tests generate a file under `<Yocto-build-dir>/tmp/log/oeqa/testresults.json`; you just pass this file as an argument, and the XML jUnit files get generated in the working directory. I've added a usage line as a comment, as @fhajredini requested.

**Pull request merged**